src/kagglehub/http_resolver.py

```diff
@@ -87,3 +137,7 @@ def _build_get_instance_url_path(h: ModelHandle) -> str:
 
 def _build_download_url_path(h: ModelHandle) -> str:
     return f"models/{h.owner}/{h.model}/{h.framework}/{h.variation}/{h.version}/download"
+
+
+def _build_dataset_download_url_path(h: DatasetHandle) -> str:
+    return f"datasets/{h.owner}/{h.dataset_name}/download"
```

Comment:
The models and datasets APIs don't always expose the same endpoints, and this doesn't look quite right to me either.  Take a look at `dataset_api_service.proto` for more info.